app/etl/item/content/parsers/mainstream.rb

```diff
@@ -0,0 +1,27 @@
+class Item::Content::Parsers::Mainstream
+  def parse(json)
+    html = []
+    html << json.dig("title") unless json.dig("title").nil?
+    html << json.dig("description") unless json.dig("description").nil?
+
+    children = json.dig("links", "children")
+    unless children.nil?
+      children.each do |child|
+        html << child["title"] unless child["title"].nil?
+      end
+    end
+
+    related_topics = json.dig("links", "related_topics")
+    unless related_topics.nil?
+      related_topics.each do |topic|
+        html << topic["title"] unless topic["title"].nil?
+      end
+    end
+
+    html.join(" ") unless html.empty?
+  end
+
+  def schemas
+    %w[mainstream_browse_page]
+  end
```

Comment:
What are your thoughts about?

- using `if` instead of `unless`
- removing `.nil?` and use `compact`
- remove `unless html.empty?`

```ruby
  def parse(json)
    html = []
    html << json.dig("title")
    html << json.dig("description")

    children = json.dig("links", "children")
    if children.present?
      children.each do |child|
        html << child["title"]
      end
    end

    related_topics = json.dig("links", "related_topics")
    if related_topics.present?
      related_topics.each do |topic|
        html << topic["title"]
      end
    end

    html.compact.join(" ")
  end
```

Reply:
@pmanrubia I agree that `parser_spec` is getting too large. I will work on this in a new PR
@MatMoore I took the view `if we can extract the data we should`, if it has no value we don't have to use it or we can remove it.